yellowbrick/regressor/residuals.py

```diff
@@ -790,15 +799,24 @@ def residuals_plot(
         **kwargs
     )
 
-    # Create the train and test splits
-    X_train, X_test, y_train, y_test = train_test_split(
-        X, y, test_size=test_size, random_state=random_state
-    )
+    # Fit the visualizer
+    viz.fit(X_train, y_train)
 
-    # Fit and transform the visualizer (calls draw)
-    visualizer.fit(X_train, y_train, **kwargs)
-    visualizer.score(X_test, y_test)
-    visualizer.finalize()
+    # Score the visualizer
+    if X_test is not None and y_test is not None:
+        viz.score(X_test, y_test)
+    elif X_test is not None or y_test is not None:
+        raise YellowbrickValueError(
+            "both X_test and y_test are required if one is specified"
+        )
+    else:
+        viz.score(X_train, y_train)
```

Comment:
Thank you for adding in this logic to protect against edge cases &mdash; this might be a useful snippet for others to use as a reference for other `ScoreVisualizer` quick methods!